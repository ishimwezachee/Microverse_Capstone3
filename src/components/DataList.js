import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Data from './Data';

const DataList = () => {
  const { country } = useParams();
  const data = useSelector((state) => state.reducer);
  return (
    <div>
      {
            data.covid
              .filter((countryData) => countryData[0] === country)
              .map((countryInfo) => <Data key={country} data={countryInfo[1]} />)
        }
    </div>
  );
};

export default DataList;
