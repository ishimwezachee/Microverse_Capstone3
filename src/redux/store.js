import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import reducer from './action-reducer';

const reducers = combineReducers({
  reducer,
});

const store = createStore(reducers, applyMiddleware(logger, thunk));

export default store;
