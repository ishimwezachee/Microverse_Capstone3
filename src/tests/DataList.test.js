import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DataList from '../components/DataList';

describe('Country Component test', () => {
  it('Renders Deatils', () => {
    const app = renderer.create(
      <Provider store={store}>
        <DataList />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
