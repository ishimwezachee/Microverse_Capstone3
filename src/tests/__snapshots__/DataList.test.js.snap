// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Component test Renders Deatils 1`] = `<div />`;
